src/frequenz/sdk/config/_config_managing.py

```diff
@@ -19,21 +20,48 @@
 
 
 class ConfigManagingActor(Actor):
-    """An actor that monitors a TOML configuration file for updates.
+    """An actor that monitors a TOML configuration files for updates.
 
-    When the file is updated, the new configuration is sent, as a [`dict`][], to the
-    `output` sender.
+    When the actor is started the configuration files will be read and sent to the
+    output sender. Then the actor will start monitoring the files for updates.
 
-    When the actor is started, if a configuration file already exists, then it will be
-    read and sent to the `output` sender before the actor starts monitoring the file
-    for updates. This way users can rely on the actor to do the initial configuration
-    reading too.
+    If no configuration file could be read, the actor will raise a exception.
+
+    The configuration files are read in the order of the paths, so the last path will
+    override the configuration set by the previous paths. Dict keys will be merged
+    recursively, but other objects (like lists) will be replaced by the value in the
+    last path.
+
+    Example:
+        If `config1.toml` contains:
+
+        ```toml
+        var1 = 1
+        var2 = 2
+        ```
+
+        And `config2.toml` contains:
+
+        ```toml
+        var2 = 3
+        var3 = 4
+        ```
+
+        Then the final configuration will be:
+
+        ```py
+        {
+            "var1": 1,
+            "var2": 3,
+            "var3": 4,
+        }
+        ```
     """
 
     # pylint: disable-next=too-many-arguments
     def __init__(
         self,
-        config_path: pathlib.Path | str,
+        config_paths: abc.Iterable[pathlib.Path | str],
```

Comment:
I wonder if it would make more sense for `config_paths` to be a `collections.abc.Sequence` rather than a `collections.abc.Iterable`, since order matters. Otherwise, the user could pass an iterable that might not preserve the expected order through the iteration

Reply:
I didn't understand very well what you mean, as an iterable will always returns items in *some order*. But then you made me think about `set`, an in this case it is not guaranteed that items are iterated in the order you created it, so I can see a point in favor of `Sequence`, as passing a set could be buggy in a obscure way. It is noisy in my head that a `Sequence` is much more restrictive than a `Iterable` (for example many iterable stuff is safe to use here, like `dict.keys()`, `map`, `filter`, etc.), but I guess if in doubt is better to go with the restrictive option and relax it in the future (forward-compatible) than the other way around. :+1:

Reply:
updated

Reply:
I had a similar reasoning about it, and I thought my comment was clear :angel: We'll provide an example next time.
I'm aware `Sequence` is more restrictive but I believe it's worth it in this case and indeed, it can be relaxed if new use-cases needed.
Thanks!